vhost_user_fs/src/passthrough.rs

```diff
@@ -641,7 +646,7 @@ impl FileSystem for PassthroughFs {
 
     fn init(&self, capable: FsOptions) -> io::Result<FsOptions> {
         // Safe because this is a constant value and a valid C string.
-        let root = unsafe { CStr::from_bytes_with_nul_unchecked(ROOT_CSTR) };
+        let root = CString::new(self.cfg.root_dir.as_str()).expect("CString::new failed");
```

Comment:
You'll also need to fix the `do_lookup` function or a client would be able to trivially break out of the shared directory by sending a lookup request with the root directory inode and name == "../".

Reply:
Yes I agree this is not secure enough but I was planning on submitting a follow up PR once this one gets merged. I was thinking about actually changing the mount namespace and performing a pivot root, which should be more secure than checking things from `do_lookup`.
Do you think this would be the best approach?

Reply:
sgtm